src/ReaderThemeSupportFeatures.php

```diff
@@ -203,13 +256,35 @@ public function update_cached_theme_support() {
 	 */
 	public function get_theme_support_features( $reduced = false ) {
 		$features = [];
+
 		foreach ( array_keys( self::SUPPORTED_FEATURES ) as $feature_key ) {
-			$feature_value = current( (array) get_theme_support( $feature_key ) );
+			if ( $this->theme_has_theme_json() ) {
+				$feature_value   = [];
+				$global_settings = wp_get_global_settings( self::SUPPORTED_THEME_JSON_FEATURES[ $feature_key ], 'theme' );
```

Comment:
While the `theme_has_theme_json` method has a check for `wp_get_global_settings`, it also now short-circuits if the `wp_theme_has_theme_json` function exists. Therefore, I think it may be safer just to move the function exists check here:

```suggestion
			if ( $this->theme_has_theme_json() && function_exists( 'wp_get_global_settings' ) ) {
				$feature_value   = [];
				$global_settings = wp_get_global_settings( self::SUPPORTED_THEME_JSON_FEATURES[ $feature_key ], 'theme' );
```

Reply:
Yes, it seems much better.

`wp_theme_has_theme_json` is introduced in WP 6.2 so `wp_get_global_settings` will exists, but this solution looks cleaner to me.